README
=======================================================
QTLMap : a software dedicated to the detection 
of QTL from experimental designs in outbred population.

Website : http://www.inra.fr/qtlmap
=======================================================

=======================================================
Prerequisites
=======================================================
- GCC, the GNU Compiler Collection (http://gcc.gnu.org/)
- LAPACK (http://www.netlib.org/lapack/)
- CMake (http://www.cmake.org/)

=======================================================
Optionals Prerequisites
=======================================================
CUDA support
- CUDA toolkit (http://developer.nvidia.com/cuda-downloads)

Developer documentation :
- ROBODoc : http://rfsber.home.xs4all.nl/Robo/

========================================================
Installation
========================================================
create a new directory inside the QTLMap directory (${QTLMAP_DIR}) 
called "build" with the following command :
mkdir build;cd build

---------------------------
** Build QTLMap with OpenMP support
cmake -DCMAKE_Fortran_COMPILER=gfortran -DCMAKE_BUILD_TYPE=Release ..

** Build QTLMap with CUDA support (Double precision)
cmake -DCMAKE_Fortran_COMPILER=gfortran -DCMAKE_BUILD_TYPE=Release -DCUDA_IMPL=true -DTYPE_ARCH_CUDA="compute_20" ../..

** Build QTLMap with CUDA support (Simple precision)
cmake -DCMAKE_Fortran_COMPILER=gfortran -DCMAKE_BUILD_TYPE=Release -DCUDA_IMPL=true -DTYPE_ARCH_CUDA="compute_20" -DCUDA_SP=true ../.. 

* CMake additionals options 
---------------------------
** LAPACK path
   -DLAPACK_LIBRARY_PATH=/path/lapack

** CUDA architecture
   -DTYPE_ARCH_CUDA="compute_X", where X=10,11,12,13,20 (see documentation of the compiler nvcc)

** ROBODoc generation
   -DROBODOC=true

** NAG ( Numerical Algorithms Group : http://www.nag.co.uk/)
  - This option replaces the call of some function using lapack/NLopt/orderpack libraries.   
     * E04JYF : quasi-Newton algorithm
     * G01EEF,G01EAF,G01FAF,G01AAF,G01ECF,G01EBF,G03ACF,G05EHF : calculation of probability 
     * M01CAF,M01DAF,M01DBF,M01DAF,M01DBF,M01DCF,M01ZAF,M01DFF : vector operations
     * F03ABF,F01ADF : matrix operations
     
  -DUSE_NAG_LIBRARY=true
  
     
execute the command "make" to compile the application 
and "make install" to execute qtlmap from ${QTLMAP_DIR}/bin.


Note : On CUDA support, only analysis : 3,4,25,26,27,28 are available

========================================================
Execution
========================================================

The examples can be find in the sample directory

========================================================
Information 
========================================================
Subscribe and post any message/question to the qtlmap-users list. 
qtlmap-users@listes.inra.fr


ChangeLog
2019-12-24 00:00  ofilangi

        * compilation with gcc suite 7.4/cmake 3.10.2
	* qtlmap 1.0.0
	
2013-09-30 13:33  ofilangi

	* src/analyse/m_qtlmap_analyse.F95,
	  src/analyse/m_qtlmap_incidence.F95: correction compilation CUDA
	  (du aux modifications des parametres d'attributs IN/OUT pour la
	  conformité avec ifort)

2013-09-30 13:21  ofilangi

	* src/data/m_qtlmap_type_cli.F95: correction affichage de l'aide.
	  Les modes verbeux et l'option print-all etait affcihé dans
	  --help-dev. deplacé dans --help

2013-09-30 13:08  ofilangi

	* src/data/m_qtlmap_type_lrt_solution.f95,
	  src/output/m_qtlmap_output_handler.f95: correction bug affichage
	  LRT

2013-09-30 12:59  ofilangi

	* src/data/m_qtlmap_type_lrt_solution.f95: correction bug index
	  chromosome

2013-09-30 12:18  ofilangi

	* src/analyse/m_qtlmap_analyse.F95,
	  src/analyse/m_qtlmap_analyse_modlin_cox.f95,
	  src/analyse/m_qtlmap_incidence.F95,
	  src/data/m_qtlmap_type_dataset.f95,
	  src/data/m_qtlmap_type_lrt_solution.f95,
	  src/data/m_qtlmap_type_parameter.f95,
	  src/output/m_qtlmap_output_handler.f95: - Modification de
	  l'affichage des LRT
	  - Supression du fichier grid_qtl (redondance avec les infos du
	  fichier LRT)

2013-09-30 11:31  ofilangi

	* src/log/m_qtlmap_log.F95: correction log (retour chariot en trop
	  pour les warning et error)

2013-09-27 09:22  ofilangi

	* src/data/m_qtlmap_type_cli.F95: mise en conformité de l'aide avec
	  la compilation ifort

2013-09-27 08:46  ofilangi

	* cmake_modules/CompileOptionsQTLMap.cmake: optimisation par
	  options de compilation

2013-09-27 08:40  ofilangi

	* src/analyse/m_qtlmap_analyse.F95,
	  src/analyse/m_qtlmap_incidence.F95,
	  src/analyse/m_qtlmap_incidence_multi.f95: correction BUG avec une
	  compilation intel.
	  Le module incidence pointe sur la structure de donnée dataset et
	  spt (contenant les pdd). Ces structures etant potentiel
	  modifiable par le module incidence, ces deux structures doivent
	  etre passe en argument inout

2013-09-26 13:54  ofilangi

	* cmake_modules/CompileOptionsQTLMap.cmake: le link avec les lib
	  gfortran et gcc devient static

2013-09-26 13:35  ofilangi

	* src/output/m_qtlmap_output_handler.f95: on enleve l'affichage du
	  seuil génomique

2013-09-26 13:26  ofilangi

	* src/qtlmap.F95: closes #2181
	  
	  On verifie qu'une famille de plein frere a une effectif superieur
	  à 10 si l'utilisateur demande une permutation pour l obtention
	  des seuils

2013-09-26 13:15  ofilangi

	* src/data/m_qtlmap_constant.f95, src/input/m_qtlmap_genealogy.f95:
	  closes #2181
	  
	  On verifie qu'une famille de plein frere a une effectif superieur
	  à 10 si l'utilisateur demande une permutation pour l obtention
	  des seuils

2013-09-26 12:26  ofilangi

	* src/output/m_qtlmap_output_handler.f95: closes #1952
	  
	  Pour chaque positions (milti qtl) on affiche le chromosome et sa
	  position successivement.

2013-09-26 12:08  ofilangi

	* src/output/m_qtlmap_output_handler.f95: closes #303
	  
	  correction du nombre d effectif affiché dans le summary

2013-09-26 11:06  ofilangi

	* src/input/m_qtlmap_phenotype.f95: correction affichage en cas d
	  erreur de lecture du fichier de genotype

2013-09-23 15:12  ofilangi

	* src/analyse/m_qtlmap_analyse.F95: correction compil CUDA

2013-09-23 15:11  ofilangi

	* lib/randlib/sdot.f, lib/randlib/spofa.f: correction
	  dimmensionnement tableau

2013-09-16 13:54  ofilangi

	* src/input/m_qtlmap_phenotype.f95: nouvel argument à l'appel
	  setgmn (randlib a été updaté)

2013-09-16 13:41  ofilangi

	* lib/randlib/phrtsd.f, lib/randlib/sdot.f, lib/randlib/setgmn.f,
	  lib/randlib/snorm.f, lib/randlib/spofa.f: update de la lib
	  randlib

2013-09-16 11:09  ofilangi

	* src/output/m_qtlmap_output_handler.f95: Modification de l'entete
	  du fichier de simulation des LRTMax
	  
	  closes #1953

2013-09-16 09:47  ofilangi

	* src/qtlmap.F95: on remet les log à INFO apres la phase de
	  simulation

2013-09-16 09:03  ofilangi

	* src/data/m_qtlmap_type_parameter.f95: moficiation de la valeur
	  par defaut de K_PROB_HAPLO_MIN
	  closes #1947

2013-09-16 08:55  ofilangi

	* src/data/m_qtlmap_type_parameter.f95: modification de la clef
	  in_race => in_pop
	  closes #1816

2013-09-13 15:22  ofilangi

	* src/output/m_qtlmap_output_handler.f95: l'option recl ne doit pas
	  etre parametrer de la meme maniere pour ifort ou gfortran.
	  creation d'un environnement INTEL_COMP et GNU_COMP

2013-09-13 15:11  ofilangi

	* testdev/execution/transcript/test_multi.sh: amelioration/bug
	  affichage du script de test

2013-09-13 15:10  ofilangi

	* testdev/execution/calcul_corcd/test.sh,
	  testdev/execution/calcul_corcd/test_multi.sh,
	  testdev/execution/ld/test.sh,
	  testdev/execution/output_files/test.sh,
	  testdev/execution/output_files/test_multi.sh,
	  testdev/execution/transcript/test.sh: amelioration/bug affichage
	  du script de test

2013-09-13 15:10  ofilangi

	* cmake_modules/CompileOptionsQTLMap.cmake: option pour ifort

2013-09-13 14:03  ofilangi

	* src/log/m_qtlmap_log.F95: couleur pour le debug et le verbose

2013-09-13 13:55  ofilangi

	* src/output/m_qtlmap_output_handler.f95: mise en conformité avec
	  le compilateur intel : recl en entrée de open ne doit pas exceder
	  2**10

2013-09-13 13:50  ofilangi

	* src/log/m_qtlmap_log.F95: correction affichage

2013-09-13 13:47  ofilangi

	* src/input/m_qtlmap_phenotype.f95: correction bug sur la lecture
	  du fichier model

2013-09-13 13:40  ofilangi

	* src/input/m_qtlmap_genotype.f95: correction bug ifort

2013-09-13 13:40  ofilangi

	* src/data/m_qtlmap_type_parameter.f95: affichage warning en trop

2013-09-13 13:40  ofilangi

	* src/log/m_qtlmap_log.F95: creation d'un fmt pour l'affichage
	  (ifort compatibilité)

2013-09-13 12:17  ofilangi

	* testdev/execution/ld, testdev/execution/ld/OUTPUT,
	  testdev/execution/ld/README, testdev/execution/ld/carte,
	  testdev/execution/ld/genea, testdev/execution/ld/model,

lib/nlopt-1.2/api/nlopt.f
      integer NLOPT_GN_DIRECT
      parameter (NLOPT_GN_DIRECT=0)
      integer NLOPT_GN_DIRECT_L
      parameter (NLOPT_GN_DIRECT_L=1)
      integer NLOPT_GN_DIRECT_L_RAND
      parameter (NLOPT_GN_DIRECT_L_RAND=2)
      integer NLOPT_GN_DIRECT_NOSCAL
      parameter (NLOPT_GN_DIRECT_NOSCAL=3)
      integer NLOPT_GN_DIRECT_L_NOSCAL
      parameter (NLOPT_GN_DIRECT_L_NOSCAL=4)
      integer NLOPT_GN_DIRECT_L_RAND_NOSCAL
      parameter (NLOPT_GN_DIRECT_L_RAND_NOSCAL=5)
      integer NLOPT_GN_ORIG_DIRECT
      parameter (NLOPT_GN_ORIG_DIRECT=6)
      integer NLOPT_GN_ORIG_DIRECT_L
      parameter (NLOPT_GN_ORIG_DIRECT_L=7)
      integer NLOPT_GD_STOGO
      parameter (NLOPT_GD_STOGO=8)
      integer NLOPT_GD_STOGO_RAND
      parameter (NLOPT_GD_STOGO_RAND=9)
      integer NLOPT_LD_LBFGS_NOCEDAL
      parameter (NLOPT_LD_LBFGS_NOCEDAL=10)
      integer NLOPT_LD_LBFGS
      parameter (NLOPT_LD_LBFGS=11)
      integer NLOPT_LN_PRAXIS
      parameter (NLOPT_LN_PRAXIS=12)
      integer NLOPT_LD_VAR1
      parameter (NLOPT_LD_VAR1=13)
      integer NLOPT_LD_VAR2
      parameter (NLOPT_LD_VAR2=14)
      integer NLOPT_LD_TNEWTON
      parameter (NLOPT_LD_TNEWTON=15)
      integer NLOPT_LD_TNEWTON_RESTART
      parameter (NLOPT_LD_TNEWTON_RESTART=16)
      integer NLOPT_LD_TNEWTON_PRECOND
      parameter (NLOPT_LD_TNEWTON_PRECOND=17)
      integer NLOPT_LD_TNEWTON_PRECOND_RESTART
      parameter (NLOPT_LD_TNEWTON_PRECOND_RESTART=18)
      integer NLOPT_GN_CRS2_LM
      parameter (NLOPT_GN_CRS2_LM=19)
      integer NLOPT_GN_MLSL
      parameter (NLOPT_GN_MLSL=20)
      integer NLOPT_GD_MLSL
      parameter (NLOPT_GD_MLSL=21)
      integer NLOPT_GN_MLSL_LDS
      parameter (NLOPT_GN_MLSL_LDS=22)
      integer NLOPT_GD_MLSL_LDS
      parameter (NLOPT_GD_MLSL_LDS=23)
      integer NLOPT_LD_MMA
      parameter (NLOPT_LD_MMA=24)
      integer NLOPT_LN_COBYLA
      parameter (NLOPT_LN_COBYLA=25)
      integer NLOPT_LN_NEWUOA
      parameter (NLOPT_LN_NEWUOA=26)
      integer NLOPT_LN_NEWUOA_BOUND
      parameter (NLOPT_LN_NEWUOA_BOUND=27)
      integer NLOPT_LN_NELDERMEAD
      parameter (NLOPT_LN_NELDERMEAD=28)
      integer NLOPT_LN_SBPLX
      parameter (NLOPT_LN_SBPLX=29)
      integer NLOPT_LN_AUGLAG
      parameter (NLOPT_LN_AUGLAG=30)
      integer NLOPT_LD_AUGLAG
      parameter (NLOPT_LD_AUGLAG=31)
      integer NLOPT_LN_AUGLAG_EQ
      parameter (NLOPT_LN_AUGLAG_EQ=32)
      integer NLOPT_LD_AUGLAG_EQ
      parameter (NLOPT_LD_AUGLAG_EQ=33)
      integer NLOPT_LN_BOBYQA
      parameter (NLOPT_LN_BOBYQA=34)
      integer NLOPT_GN_ISRES
      parameter (NLOPT_GN_ISRES=35)
      integer NLOPT_FAILURE
      parameter (NLOPT_FAILURE=-1)
      integer NLOPT_INVALID_ARGS
      parameter (NLOPT_INVALID_ARGS=-2)
      integer NLOPT_OUT_OF_MEMORY
      parameter (NLOPT_OUT_OF_MEMORY=-3)
      integer NLOPT_ROUNDOFF_LIMITED
      parameter (NLOPT_ROUNDOFF_LIMITED=-4)
      integer NLOPT_SUCCESS
      parameter (NLOPT_SUCCESS=1)
      integer NLOPT_MINF_MAX_REACHED
      parameter (NLOPT_MINF_MAX_REACHED=2)
      integer NLOPT_FTOL_REACHED
      parameter (NLOPT_FTOL_REACHED=3)
      integer NLOPT_XTOL_REACHED
      parameter (NLOPT_XTOL_REACHED=4)
      integer NLOPT_MAXEVAL_REACHED
      parameter (NLOPT_MAXEVAL_REACHED=5)
      integer NLOPT_MAXTIME_REACHED
      parameter (NLOPT_MAXTIME_REACHED=6)
